src/nifreeze/data/filtering.py

```diff
@@ -96,3 +102,122 @@ def advanced_clip(
         data = np.round(255 * data).astype(dtype)
 
     return data
+
+
+def detrend_data_percentile(data: np.ndarray, mask: np.ndarray | None = None) -> np.ndarray:
```

Comment:
Perhaps a more representative name and access to the percentiles?
```suggestion
def robust_minmax_normalization(data: np.ndarray, mask: np.ndarray | None = None, pmin: float = 5.0, pmax: float = 95.0) -> np.ndarray:
```